pkg/binlogfile/binlogger.go

```diff
@@ -187,6 +191,53 @@ func (b *binlogger) ReadFrom(from binlog.Pos, nums int32) ([]binlog.Entity, erro
 	return ents, err
 }
 
+// ReadAll reads all binlog in the directory.
+func (b *binlogger) ReadAll(stop chan struct{}) (result chan *binlog.Entity, errChan chan error) {
+	errChan = make(chan error, 1)
+	result = make(chan *binlog.Entity, 1)
+
+	minSuffix := uint64(math.MaxUint64)
+	names, err := ReadBinlogNames(b.dir)
+	if err != nil {
+		log.Error("read binlog files failed", zap.Error(err))
+	} else {
+		// Find the minimal suffix so we can construct start pos.
+		for _, name := range names {
+			suffix, _, err := ParseBinlogName(name)
+			if err != nil {
+				break
+			}
+			if suffix < minSuffix {
+				minSuffix = suffix
+			}
+		}
+	}
+	if err != nil || len(names) == 0 {
+		errChan <- err
+		close(errChan)
+		close(result)
+		return
+	}
+
+	from := binlog.Pos{Suffix: minSuffix, Offset: 0}
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
If you use `context.Context` instead of `stop` as the argument, we can simplify the following `Walk` as:

```go
select {
case <-ctx.Done():
    return ctx.Err()
case result <- entity:
    return nil
}
```